spring-cloud-stream/src/main/java/org/springframework/cloud/stream/annotation/rxjava/RxJavaProcessor.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.annotation.rxjava;
+
+import rx.Observable;
+
+/**
+ * Marker interface that RxJava module uses to provide the implementation bean.
+ *
+ * @author Mark Pollack
+ * @author Ilayaperumal Gopinathan
+ */
+public interface RxJavaProcessor<I, O> {
+
+	Observable<O> process(Observable<I> input);
```

Comment:
@ericbottard I still think that the Rx programming style should apply, but on a second thought, let's leave it as it is. I'll try to answer that question in a more elaborate way. It was related to the idea of providing bindable `@Input`s and `@Output`s that are Observable(s) and, respectively, Subject(s) - case in which they would be created by the framework so a processor could be

   void process(Observable<I> input, Subject<?,O> output)  {
     input.map(something).subscribe(output);
   } 

which is still fairly idiomatic, as far as I can tell. Obviously, the last `subscribe` is boilerplate in the 1:1 transformation case of the Processor, but would be applicable for multiple outputs, etc. Anyway, I think this warrants a separate discussion.